### PR TITLE
[1LP][RFR] Update release script

### PR DESCRIPTION
**Pull request by mshriver**
Refactor the release script a bit.

Don't use local repo for commit walking
Filter PRs on date range between given releases
Use tabulate to format the output, no more length calculations and handling
Include author and multiple labels in the output

Release 17.63.0 has been updated with the output from the script while including `--brief --links`.
https://github.com/ManageIQ/integration_tests/releases/tag/17.63.0

Example stats output, in terminal the columns are aligned.
```
Report Includes: 17.62.0 -> 17.63.0
Label                  Number of PRs
-------------------  ---------------
fix-test                          20
test-automation                   15
enhancement                       10
rc-regression-fix                  8
fix-framework                      7
new-test-or-feature                5
test-cleanup                       5
fix-locator-or-text                3
other                              3
manual                             3
customer-case                      2
tech-debt                          2
issue-bug                          1
requirements                       1
py3-compat                         1
Author             Number of PRs
---------------  ---------------
izapolsk                      11
digitronik                     9
mshriver                       8
john-dupuy                     6
sshveta                        5
jawatts                        4
ganeshhubale                   4
niyazRedhat                    4
psimovec                       3
sbulage                        3
lina-nikiforova                2
mnadeem92                      2
jaryn                          2
valaparthvi                    2
prichard77                     2
tpapaioa                       1
nachandr                       1
mmojzis                        1
```

### CODE REVIEW

File: cfme/scripting/release.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+)
+def main(tag, old_tag, report_type, links, tableformat):
+    """Script to assist in generating the release changelog
+
+    This script will generate a simple or full diff of PRs that are merged
+    and present the data in a way that is easy to copy/paste into an email
+    or git tag.
+    """
+
+    print(f"Report Includes: {old_tag} -> {tag}")
```
Comment (john-dupuy) [inline cfme/scripting/release.py]: _Suggestion_ I think we should use `click.echo` rather than `print`

File: cfme/scripting/release.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+        # Label stats
+        print(tabulate.tabulate(sorted(labels.items(),
+                                       key=lambda item: item[1],
+                                       reverse=True),
+                                headers=["Label", "Number of PRs"],
+                                tablefmt=tableformat))
+
+        print('======================================')
+
+        # Author stats
```
Comment (john-dupuy) [inline cfme/scripting/release.py]: Anyway to get `Reviewer stats`? Definitely could/would be a separate PR

File: cfme/scripting/release.py [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    "sprout",
+    "tech-debt",
+    "test-automation",
+    "test-cleanup",
+    "widgetastic-conversion",
+]
+
+IGNORED_LABELS = ['lint-ok', 'WIP-testing']
+
+PR_LINK = "[{pr}](https://github.com/ManageIQ/integration_tests/pull/{pr})"
```
Comment (john-dupuy) [inline cfme/scripting/release.py]: I think this should use `REPO_NAME` in its definition.

File: cfme/scripting/release.py [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+    "rc-regression-fix",
+    "Redfish",
+    "requirements",
+    "RHV",
+    "sprout",
+    "tech-debt",
+    "test-automation",
+    "test-cleanup",
+    "widgetastic-conversion",
+]
```
Comment (john-dupuy) [inline cfme/scripting/release.py]: Is there anyway to fetch these via the GH api? I see there is `repo.get_labels()`, this returns a paginated list of GH label objects. Hmm, but you still want a list of `valid_labels`. Could we get away with just a list of `invalid_labels`?

File: cfme/scripting/release.py (hunk 2 above)
Comment (mshriver) [reply cfme/scripting/release.py]: Yes and yes :)

File: cfme/scripting/release.py (hunk 3 above)
Comment (mshriver) [reply cfme/scripting/release.py]: talked about it offline, not worth it for the messiness of two-stage formatting.

File: cfme/scripting/release.py (hunk 4 above)
Comment (mshriver) [reply cfme/scripting/release.py]: We can get them via the API, this is kind of filtering out `RFR`, `1LP`, `WIPTEST` kind of labels.

I think its worth further discussion, okay with it being in a followup (along with additional stats?)

Comment (john-dupuy) [reply cfme/scripting/release.py]: Yep, a followup is cool with me
